cmd/cluster-agent/subcommands/start/command.go

```diff
@@ -113,6 +113,12 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/collector/python"
 )
 
+// This hack allows us to retrieve these fields with `fx.Provide()`
```

Comment:
I don't think these should be made into components, since the goal isn't to create reusable pieces of code. Instead we just want to leverage fx's ability to coordinate component creation and dependencies. With that in mind, I would turn these into structs:

```
type clusterInfo struct {
  Name string
  ID string
}
type hostInfo struct {
  Hostname string
}
```

And use those in the `fx.Provide` calls below. This is a much more standard way to use fx.

Reply:
Thank you for the proposal. I applied the recommendation.